docs/reference/filter-reference.md

```diff
@@ -0,0 +1,100 @@
+# Filters
+
+Filters are used to extend Ambassador to run additional custom logic on each request. Filters are writting in Golang.
+
+## Filter Definition
+
+Filters are registered with the `Filter` custom resource definition.
+
+```yaml
+---
+apiVersion: getambassador.io/v1beta1
+kind: Filter
+metadata:
+  name: param-filter
+  namespace: standalone
+spec:
+  Plugin:
+    name: param-filter # The plugin's `.so` file's base name
+```
+
+## Filter Configuration
+
+The `FilterPolicy` custom resource definition specifies which filters are run, and when. In the example below, the `param-filter` registered above is run on requests to `/test/` and `/httpbin/`. 
+
+```yaml
+---
+apiVersion: getambassador.io/v1beta1
+kind: Policy
+metadata:
+  name: httpbin-policy
+  namespace: default
+spec:
+  # everything defaults to private; you can create rules to make stuff
+  # public, and you can create rules to require additional scopes
+  # which will be automatically checked
+  rules:
+  - host: "*"
+    path: /test/*
+    public: false # must be false if using a middleware
```

Comment:
As of https://github.com/datawire/apro/pull/108 (not yet merged) `public` no longer exists; you can explicitly say that something is public by writing `filters: null`, but otherwise setting `filters:` implies `public: true`.